qiskit/pulse/commands/sample_pulse.py

```diff
@@ -151,7 +151,9 @@ def __hash__(self):
         return hash((super().__hash__(), self.samples.tostring()))
 
     def __repr__(self):
-        return '%s(%s, duration=%d)' % (self.__class__.__name__, self.name, self.duration)
+        return '%s(samples=%s, name="%s")' % (self.__class__.__name__,
```

Comment:
`samples` is better to be truncated at some point to handle full resolution api and I think `duration` is still convenient to know the pulse length: something like `samples=[0+0j, 1+0j, ...] (150 dt), name=`

Reply:
> s still convenient to know the pulse length: something like samples=[0+0j, 1+0j, ...] (150 dt), name=

No, this violates the [python datamodel](https://docs.python.org/3/reference/datamodel.html#object.__repr__). This would be the role of `__str__` which could be in a separate PR.

Reply:
> samples is better to be truncated at some point to handle full resolution 
This can be done with the global `np.set_printoptions` I was debating if I should set this or let the global numpy settings handle it.

Reply:
fixed truncation